Unable to build native image on Macbook Pro M4 Pro

### Describe the bug

I'm trying to create a native image using Apple M4 Pro but I haven't been able to do it due to an error when getting the graalvm version.


`./gradlew -Dquarkus.package.type=native -Dquarkus.native.container-build=true -Dquarkus.container-image.build=true clean build`

```
> There was a failure while executing work items
   > A failure occurred while executing io.quarkus.gradle.tasks.worker.BuildWorker
      > io.quarkus.builder.BuildException: Build failure: Build failed due to errors
                [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to get GraalVM version
                at io.quarkus.deployment.pkg.steps.NativeImageBuildRunner.getGraalVMVersion(NativeImageBuildRunner.java:41)
                at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:236)
                at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
                at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
                at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
                at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
                at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
                at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
                at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
                at java.base/java.lang.Thread.run(Thread.java:833)
                at org.jboss.threads.JBossThread.run(JBossThread.java:483)
        Caused by: java.lang.IllegalArgumentException: Cannot parse version from output:
        #
        # A fatal error has been detected by the Java Runtime Environment:
        #
        #  SIGILL (0x4) at pc=0x0000ffff6bd3fc5c, pid=1, tid=15
        #
        # JRE version:  (21.0.5+11) (build )
        # Java VM: OpenJDK 64-Bit Server VM (21.0.5+11-LTS, mixed mode, tiered, jvmci, compressed oops, compressed class ptrs, g1 gc, linux-aarch64)
        # Problematic frame:
        # j  java.lang.System.registerNatives()V+0 java.base
        #
        # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
        #
        # An error report file with more information is saved as:
        # /project/hs_err_pid1.log
        [0.011s][warning][os] Loading hsdis library failed
        #
        # The crash happened outside the Java Virtual Machine in native code.
        # See problematic frame for where to report the bug.
        #

        [error occurred during error reporting (), id 0x5, SIGTRAP (0x5) at pc=0x0000ffff84783afc]
                at io.quarkus.deployment.pkg.steps.GraalVM$Version.of(GraalVM.java:327)
                at io.quarkus.deployment.pkg.steps.NativeImageBuildRunner.getGraalVMVersion(NativeImageBuildRunner.java:38)
                ... 10 more
```

### Expected behavior

I expect to be able to include custom parameters that are passed to "docker run" when I want to generate a native image.

`docker run --env LANG=C --env _JAVA_OPTIONS=-XX:UseSVE=0 --rm --user 1000:1000 -i quay.io/quarkus/ubi-quarkus-mandrel-builder-image:jdk-21 --version
`


### Actual behavior

Under the hood Quarkus executes this command

`docker run --env LANG=C --rm --user 1000:1000 -i quay.io/quarkus/ubi-quarkus-mandrel-builder-image:jdk-21 --version`

I know that the solution for this is to add  [_JAVA_OPTIONS=-XX:UseSVE=0 ]( https://github.com/corretto/corretto-21/issues/85#issuecomment-2605714791) but it is not possible to do so because [NativeImageBuildRunner.getGraalVMVersion/NativeImageBuildContainerRunner.buildCommand](https://github.com/quarkusio/quarkus/blob/202c65c50abeb02a0d196e63d3be7ff996951467/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildContainerRunner.java#L120-L123) does not receive additional parameters through environment variables.

This method has the problem
https://github.com/quarkusio/quarkus/blob/202c65c50abeb02a0d196e63d3be7ff996951467/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildContainerRunner.java#L120-L123

This method is already working well https://github.com/quarkusio/quarkus/blob/202c65c50abeb02a0d196e63d3be7ff996951467/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildContainerRunner.java#L125-L132

### How to Reproduce?

Can be reproduced by running this command

`./gradlew -Dquarkus.package.type=native -Dquarkus.native.container-build=true -Dquarkus.container-image.build=true clean build
`


### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Mandrel or GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

Hi @zakkak, I'm getting the same error.

In order to understand, and subsequently be able to generate the native image, I renamed the docker binary and replaced it with a script.


After adding the parameter you indicate, I still get the same error 

```
 Caused by: java.lang.IllegalArgumentException: Cannot parse version from output: 
        Running the command: run --env LANG=C --rm --user 0 --rm quay.io/quarkus/ubi-quarkus-mandrel-builder-image:jdk-21 --version
        #
        # A fatal error has been detected by the Java Runtime Environment:
        #
        #  SIGILL (0x4) at pc=0x0000ffff87d3fc5c, pid=1, tid=15
        #
        # JRE version:  (21.0.5+11) (build )
        # Java VM: OpenJDK 64-Bit Server VM (21.0.5+11-LTS, mixed mode, tiered, jvmci, compressed oops, compressed class ptrs, g1 gc, linux-aarch64)
        # Problematic frame:
        # j  java.lang.System.registerNatives()V+0 java.base
```


```
#!/bin/bash

if [[ "$1" == "run" ]]; then
  echo "Running the command: "$1" "${@:2}""

  # this fails
  /usr/local/bin/docker_original "$1" "${@:2}"

  # this allows me to generate the native image.
  #/usr/local/bin/docker_original "$1" --env _JAVA_OPTIONS=-XX:UseSVE=0 "${@:2}"
elif [[ "$1" == "build" ]]; then
  /usr/local/bin/docker_original "$1" "${@:2}"
else
  /usr/local/bin/docker_original "$@"
fi```

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildContainerRunner.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildContainerRunner.java
@@ -119,7 +119,16 @@ public abstract class NativeImageBuildContainerRunner extends NativeImageBuildRu
 
     @Override
     protected String[] getGraalVMVersionCommand(List<String> args) {
-        return buildCommand("run", Collections.singletonList("--rm"), args);
+        List<String> containerRuntimeArgs;
+        if (nativeConfig.containerRuntimeOptions().isPresent()) {
+            List<String> runtimeOptions = nativeConfig.containerRuntimeOptions().get();
+            containerRuntimeArgs = new ArrayList<>(runtimeOptions.size() + 1);
+            containerRuntimeArgs.addAll(runtimeOptions);
+            containerRuntimeArgs.add("--rm");
+        } else {
+            containerRuntimeArgs = Collections.singletonList("--rm");
+        }
+        return buildCommand("run", containerRuntimeArgs, args);
     }
 
     @Override

